syntax = "proto3";

package secshrnet;

message Share {
    uint32 index = 1;
    bytes key_share = 2;
    bytes ciphertext = 3;
    bytes ciphertext_hash = 4;
}

enum PacketType {
    STORE_SHARE = 0;
    RECOVER_SHARE = 1;
    RETURN_SHARE = 2;
    NO_SHARE = 3;
}

message Packet {
    PacketType type = 1;
    string sender = 2;
    string tag = 4;
    Share share = 5;
}
